synapse/rest/client/v2_alpha/account.py

```diff
@@ -65,9 +94,90 @@ def on_POST(self, request):
         if existingUid is None:
             raise SynapseError(400, "Email not found", Codes.THREEPID_NOT_FOUND)
 
-        ret = yield self.identity_handler.requestEmailToken(**body)
+        if self.config.email_enable_password_reset_from_is:
+            if 'id_server' not in body:
+                raise SynapseError(400, "Missing 'id_server' param in body")
+
+            # Have the identity server handle the password reset flow
+            ret = yield self.identity_handler.requestEmailToken(**body)
```

Comment:
I'd prefer if we explicitly pulled stuff out of the dict that we need, it makes it easier to validate and see what's going on